src/greeter/GreeterApp.h

```diff
@@ -50,6 +50,9 @@ namespace SDDM {
     private slots:
         void addViewForScreen(QScreen *screen);
         void removeViewForScreen(QQuickView *view);
+        void pamShow();
+        void pamShowCloseGreeter();
+        void pamStoreMessage(const QString message);
```

Comment:
(const QString &message)

otherwise we make a copy of the QString object when we pass it to the method.
It's a pretty cheap shallow-copy, but still, may as well avoid it whilst we can.